src/main.rs

```diff
@@ -9,21 +10,34 @@ use libazureinit::distro::{Distribution, Distributions};
 use libazureinit::{
     error::Error as LibError,
     goalstate, imds, media,
+    media::Media,
     reqwest::{header, Client},
     user,
 };
 
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
-fn get_username(imds_body: String) -> Result<String, LibError> {
+fn get_username(imds_body: String) -> Result<String, anyhow::Error> {
     if imds::is_password_authentication_disabled(&imds_body)? {
         // password authentication is disabled
-        imds::get_username(imds_body.clone())
+        Ok(imds::get_username(imds_body.clone())?)
     } else {
         // password authentication is enabled
+        let mount_media = Media::new(
+            PathBuf::from(media::PATH_MOUNT_DEVICE),
+            PathBuf::from(media::PATH_MOUNT_POINT),
+        );
+        let mounted = mount_media
+            .mount()
+            .with_context(|| "Failed to mount media.")?;
+
         let ovf_body = media::read_ovf_env_to_string()?;
```

Comment:
You could move this function to the `impl Media<Mounted>` so you can't call it without the media being mounted.

Reply:
Thanks, done.